### PR TITLE
Define response headers behavior - old

### PR DESCRIPTION
**Pull request by SergeyKanzhelev**
Initial set of behavior definition. Mostly to address #50

### CODE REVIEW

File: abstract/trace-context-overview.md [hunk 1]
```diff
@@ -1,7 +1,11 @@
 # Overview
 
-Trace context represented by set of name/value pairs describing identity of every http request. As a performance optimization measure few of these pairs were promoted to a separate header. This header has fixed length and defined sequence of fields.
+Trace context represented by set of name/value properties describing identity of every http request. Few of these properties are promoted to a separate header. It also allows to align different tracing frameworks on semantic of these properties. This is also done as a performance optimization measure - header has fixed length and defined sequence of fields. It is easy to parse and mutate this header.
```
Comment (codefromthecrypt) [inline abstract/trace-context-overview.md]: Stab at a different overview:

The trace context identifies an http request and its position in a potentially distributed operation. Its properties exist in a fixed-length generic header, `Trace-Context`, and possibly extend into a vendor-specific `Trace-Context-Ext` header. The `Trace-Context` defines a sequence of fields of fixed length all vendors must process, such as a trace ID. If a vendor cannot process `Trace-Context-Ext`, it must restart the trace. The following elaborates format details and propagation semantics implied by this standard.

File: abstract/trace-context-overview.md [hunk 2]
```diff
@@ -1,7 +1,13 @@
...
 
-**TODO:** Add details on behavior when one of the headers cannot be parsed
+Libraries and platforms SHOULD strive to minimize the number and size of trace context properties. Shortening size of properties allows libraries implementing different scheme of trace correlation to append properties needed for that library. So different vendors may co-exist in the different components of the same distributed trace.
+
+Headers `Trace-Context` and `Trace-Context-Ext` can be used to pass http request identity information as the request and response headers. Libraries and platforms MAY mutate these headers according to the semantics described in corresponding sections. Libraries and platforms MUST propagate `Trace-Context` and `Trace-Context-Ext` request headers to all outgoing calls. Propagation of the header guarantees that trace correlation is not broken. Libraries and platforms SHOULD propagate `Trace-Context` and `Trace-Context-Ext` response headers.
```
Comment (yurishkuro) [inline abstract/trace-context-overview.md]: "propagate response headers" sounds wrong. The typical behavior of a server would be to send back the same trace context as received in the incoming request, not propagate something that came from downstream service,

File: abstract/trace-context-overview.md [hunk 3]
```diff
@@ -1,7 +1,13 @@
...
+
+Headers `Trace-Context` and `Trace-Context-Ext` may appear on request and response independently. Missing `Trace-Context-Ext` header means that there are no additional properties defined in the request. Missing `Trace-Context` header means that library or platform has not started the trace. However has some tracing related information that identifies the request. See corresponding section for the examples of properties that can be used in `Trace-Context-Ext` header.
```
Comment (yurishkuro) [inline abstract/trace-context-overview.md]: "However has some tracing related information" - what did you mean?

File: trace_context/HTTP_HEADER_FORMAT.md [hunk 4]
```diff
@@ -1,7 +1,7 @@
 # Trace Context HTTP Header Format
 
-A trace context header is used to pass trace context information across systems for a HTTP 
-request. Our goal is to share this with the community so that various tracing and diagnostics 
+A trace context header is used to pass trace context information across systems for an HTTP
```
Comment (yurishkuro) [inline trace_context/HTTP_HEADER_FORMAT.md]: I would avoid providing justifications here to remove redundancy. The purpose of the header is already explained in the common section. Enough to say that this part defines the format of the header specifically for HTTP protocol.

File: abstract/trace-context-overview.md (hunk 2 above)
Comment (SergeyKanzhelev) [reply abstract/trace-context-overview.md]: Clarified. I think

**Comment by bogdandrutu** [comment]
Can you please resync?

**Comment by AloisReitbauer** [comment]
Overall, we need to work on a more precise definition of a processing model anyways. I propose to commit the current state with a todo line on what needs to be still discussed. 

Here is an example of a processing model: https://www.w3.org/TR/beacon/#sec-processing-model

**Comment by AloisReitbauer** [comment]
What should we do with this PR?

**Pull request closed without merging**